src/test/hooks.ts

```diff
@@ -0,0 +1,629 @@
+
+                     throw new Error('Error in onResult')
+                  },
+               }
+
+               const hooks_creation_methods = [H.newTiny({ hooks: hooks }), tiny.withHooks(hooks)]
+
+               for (const my_tiny of hooks_creation_methods) {
+                  final_context = null
+
+                  await my_tiny.sql('a.select')
+
+                  expect(final_context).to.deep.equal(original_context)
+               }
+            })
+         })
+
+         describe('and a single non-pre hook throws an error', () => {
+            it('should catch the error and continue with the rest of the hooks', async () => {
+               let final_context: any
+
+               const hooks: TinyHooks = {
+                  preSql: (ctx, name, params) => {
+                     return { ctx: { ...ctx, foo: 'bar' }, args: [name, params] }
+                  },
+                  onQuery(_ctx, _query_begin_context) {
+                     throw new Error('Error in onQuery')
+                  },
+                  onSubmit(ctx, _query_submit_context) {
+                     return {
+                        ...ctx,
+                        onSubmit: 'made it',
+                     }
+                  },
+                  onResult(ctx, query_complete_context) {
+                     final_context = {
+                        ...ctx,
+                        ...query_complete_context,
+                     }
+
+                     return final_context
+                  },
+               }
+
+               const hooks_creation_methods = [H.newTiny({ hooks: hooks }), tiny.withHooks(hooks)]
+
+               for (const my_tiny of hooks_creation_methods) {
+                  final_context = null
+
+                  await my_tiny.sql('a.select')
+
+                  expect(final_context.onSubmit).to.equal('made it')
+               }
+            })
+         })
+      })
+
+      describe('name modification', () => {
+         it('should still run the original db_call', async () => {
+            let final_context: any
+
+            const hooks: TinyHooks = {
+               preSql: (ctx, name, params) => {
+                  return { ctx: { ...ctx, foo: 'bar' }, args: [`${name}_foo`, params] }
+               },
+               onResult(ctx, query_complete_context) {
+                  final_context = {
+                     ...ctx,
+                     ...query_complete_context,
+                  }
+
+                  return final_context
+               },
+            }
+
+            const hooks_creation_methods = [H.newTiny({ hooks: hooks }), tiny.withHooks(hooks)]
+
+            for (const my_tiny of hooks_creation_methods) {
+               final_context = null
+
+               await my_tiny.sql('a.select')
+
+               expect(final_context.query_id).to.exist
+               expect(final_context.foo).to.equal('bar')
+            }
+         })
+      })
+   })
+
+   describe('raw query', () => {
```

Comment:
there is no "raw query modification" type test for a hook that modifies the raw query in `preRawQuery` 

From the code, it looks to be the intent that it _can_ be modified. I'm curious how this relates to to the `preSql` version of a hook.